### ISSUE TITLE
📝 Syntax error in formatted code when wrapping function args with arrowParentheses: asNeeded

### ISSUE THREAD

**Issue by smably** [biomejs/biome#1428]
### Environment information

```bash
CLI:
  Version:                      1.4.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.9.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{
	"javascript": {
		"formatter": {
			"arrowParentheses": "asNeeded"
		}
	}
}
```

### Playground link

https://biomejs.dev/playground/?arrowParentheses=as-needed&code=YwBvAG4AcwB0ACAAZgBvAG8AIAA9ACAAYgBhAHIAIAA9AD4AIABgAFQAaABpAHMAIABpAHMAIABhACAAcwB0AHIAaQBuAGcAIAB0AGgAYQB0ACcAcwAgAGwAbwBuAGcAIABlAG4AbwB1AGcAaAAgAHQAbwAgAHcAcgBhAHAAIABhAG4AZAAgAGMAbwBuAHQAYQBpAG4AcwAgAGEAbgAgAGkAbgB0AGUAcgBwAG8AbABhAHQAZQBkACAAdgBhAGwAdQBlACAAJAB7AGIAYQByAH0ACgBhAG4AZAAgAHQAaABlAG4AIABhACAAbgBlAHcAbABpAG4AZQBgAA%3D%3D

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by faultyserver** [comment]
I believe this is a duplicate of #1106, and will be fixed in #1449 (added this playground example as a test case there and it is resolved by it).

**Comment by smably** [comment]
Oh, fantastic! Thank you for the speedy fix.
